source/RoslynAnalyzers/Descriptors.cs

```diff
@@ -147,21 +147,15 @@ public static DiagnosticDescriptor Oct2008IntegrationTestForwardCancellationToke
             DiagnosticSeverity.Error,
             true);
 
-        public static readonly DiagnosticDescriptor OptionalPropertiesOnMessageTypesMustBeNullable = new(
+        public static readonly DiagnosticDescriptor OptionalPropertiesOnMessageTypesMustBeInitializedOrNullable = new(
             "OCT3008",
-            "Optional Properties on MessageTypes must be nullable",
-            "Property \"{0}\" should be of type {1}? (Optional Properties on MessageTypes must be nullable)",
+            "Optional Properties on MessageTypes must be nullable or have initializers",
+            "Property \"{0}\" should have initializer or be of type {1}? (Optional Properties on MessageTypes must be initialized or nullable)",
             Category,
             DiagnosticSeverity.Error,
             true);
 
-        public static readonly DiagnosticDescriptor MessageTypesMustInstantiateCollections = new(
-            "OCT3009",
-            "MessageTypes must instantiate non-nullable collections",
-            "MessageTypes must instantiate non-nullable collections.",
-            Category,
-            DiagnosticSeverity.Error,
-            true);
+        // OCT3009 was "MessageTypes must instantiate non-nullable collections", but this is now subsumed by our "null or initialized" convention. The ID is free for something else to use
```

Comment:
I'd vote to keep this here, but mark as `[Obsolete]`.
Mainly so we can cover the use case where someone has left a comment around in a spot we weren't expecting.

Reply:
Done